modules/restore-replace-stopped-baremetal-etcd-member.adoc

```diff
@@ -0,0 +1,481 @@
+----
+<1> The new machine, `clustername-8qw5l-master-3` is being created and is ready once the phase changes from `Provisioning` to `Running`.
++
+It might take a few minutes for the new machine to be created. The etcd cluster Operator will automatically sync when the machine or node returns to a healthy state.
+
+.Verification
+
+. Verify that all etcd pods are running properly.
++
+In a terminal that has access to the cluster as a `cluster-admin` user, run the following command:
++
+[source,terminal]
+----
+$ oc get pods -n openshift-etcd | grep -v etcd-quorum-guard | grep etcd
+----
++
+.Example output
+[source,terminal]
+----
+etcd-ip-10-0-133-53.ec2.internal                 3/3     Running     0          7m49s
+etcd-ip-10-0-164-97.ec2.internal                 3/3     Running     0          123m
+etcd-ip-10-0-154-204.ec2.internal                3/3     Running     0          124m
+----
++
+If the output from the previous command only lists two pods, you can manually force an etcd redeployment. In a terminal that has access to the cluster as a `cluster-admin` user, run the following command:
++
+[source,terminal]
+----
+$ oc patch etcd cluster -p='{"spec": {"forceRedeploymentReason": "recovery-'"$( date --rfc-3339=ns )"'"}}' --type=merge <1>
+----
+<1> The `forceRedeploymentReason` value must be unique, which is why a timestamp is appended.
+
+. Verify that there are exactly three etcd members.
+
+.. Connect to the running etcd container, passing in the name of a pod that was not on the affected node:
++
+In a terminal that has access to the cluster as a `cluster-admin` user, run the following command:
++
+[source,terminal]
+----
+$ oc rsh -n openshift-etcd etcd-ip-10-0-154-204.ec2.internal
+----
+
+.. View the member list:
++
+[source,terminal]
+----
+sh-4.2# etcdctl member list -w table
+----
++
+.Example output
+[source,terminal]
+----
++------------------+---------+------------------------------+---------------------------+---------------------------+
+|        ID        | STATUS  |             NAME             |        PEER ADDRS         |       CLIENT ADDRS        |
++------------------+---------+------------------------------+---------------------------+---------------------------+
+| 5eb0d6b8ca24730c | started |  ip-10-0-133-53.ec2.internal |  https://10.0.133.53:2380 |  https://10.0.133.53:2379 |
+| 757b6793e2408b6c | started |  ip-10-0-164-97.ec2.internal |  https://10.0.164.97:2380 |  https://10.0.164.97:2379 |
+| ca8c2990a0aa29d1 | started | ip-10-0-154-204.ec2.internal | https://10.0.154.204:2380 | https://10.0.154.204:2379 |
++------------------+---------+------------------------------+---------------------------+---------------------------+
+----
++
+If the output from the previous command lists more than three etcd members, you must carefully remove the unwanted member.
++
+[WARNING]
+====
+Be sure to remove the correct etcd member; removing a good etcd member might lead to quorum loss.
+====
```

Comment:
the following verification can be added:
Validate that all etcd members are healthy 
```
sh-4.4# etcdctl endpoint health --cluster 
https://10.46.29.130:2379 is healthy: successfully committed proposal: took = 8.973065ms
https://10.46.29.129:2379 is healthy: successfully committed proposal: took = 11.559829ms
https://10.46.29.131:2379 is healthy: successfully committed proposal: took = 11.665203ms

```



Validate that all nodes are at the latest revision
```
$ oc get etcd -o=jsonpath='{range.items[0].status.conditions[?(@.type=="NodeInstallerProgressing")]}{.reason}{"\n"}{.message}{"\n"}'
AllNodesAtLatestRevision
```

Reply:
Done, thank you!